charts/fluentd/templates/hpa.yaml

```diff
@@ -0,0 +1,28 @@
+{{- if and  ( eq  .Values.kind "Deployment" )  .Values.autoscaling.enabled  }}
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "fluentd.fullname" . }}
+  labels:
+    {{- include "fluentd.labels" . | nindent 4 }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "fluentd.fullname" . }}
+  minReplicas: {{ .Values.autoscaling.minReplicas }}
+  maxReplicas: {{ .Values.autoscaling.maxReplicas }}
+  metrics:
```

Comment:
My idea here with 'extra rule' is to enable more elaborate scaling
check: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#metricspec-v2beta2-autoscaling

For example, i am using prometheus adapter and i can use custom.metricks.k8s.io to scale with custom metrics or other advanced metrics that the simple k8s metrics dont offer (for example, size of input buffer)

Reply:
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-multiple-metrics-and-custom-metrics

Reply:
We can add this after, also, so not a blocker on my part

Reply:
good feedback implemented both the new HPA API version and custom rules and custom behavior settings 👍